common/views/components/DefaultPageLayout/DefaultPageLayout.js

```diff
@@ -190,82 +123,78 @@ type Props = {|
   isPreview?: boolean
 |}
 
-const DefaultPageLayout = ({
-  children,
-  type,
-  url,
-  title,
-  description,
-  imageUrl,
-  siteSection,
-  analyticsCategory,
-  featuresCohort = 'default',
-  featureFlags = [],
-  isPreview = false
-}: Props) => (
-
-  <div>
-    <Head>
-      <meta charSet='utf-8' />
-      {/* TODO: use flag as to whether to include this */}
-      <Analytics
-        category={analyticsCategory}
-        seriesUrl={null}
-        positionInSeries={null}
-        contentType={null} />
-
-      <meta httpEquiv='X-UA-Compatible' content='IE=edge,chrome=1' />
-      <title>{`${title} | Wellcome Collection`}</title>
-      <meta name='viewport' content='width=device-width, initial-scale=1' />
-      <meta name='theme-color' content='#000000'/>
-
-      <OpenGraph
-        type={type}
-        url={url}
-        title={title}
-        description={description}
-        imageUrl={imageUrl}
-      />
-      <TwitterCard
-        type={type}
-        url={url}
-        title={title}
-        description={description}
-        imageUrl={imageUrl} />
+class DefaultPageLayout extends Component<Props> {
+  componentDidMount() {
+    const { analyticsCategory, featuresCohort } = this.props;
+    analytics({analyticsCategory, featuresCohort});
+  }
 
-      <link rel='apple-touch-icon' sizes='180x180' href='https://i.wellcomecollection.org/assets/icons/apple-touch-icon.png' />
-      <link rel='shortcut icon' href='https://i.wellcomecollection.org/assets/icons/favicon.ico' type='image/ico' />
-      <link rel='icon' type='image/png' href='https://i.wellcomecollection.org/assets/icons/favicon-32x32.png' sizes='32x32' />
-      <link rel='icon' type='image/png' href='https://i.wellcomecollection.org/assets/icons/favicon-16x16.png' sizes='16x16' />
-      <link rel='manifest' href='https://i.wellcomecollection.org/assets/icons/manifest.json' />
-      <link rel='mask-icon' href='https://i.wellcomecollection.org/assets/icons/safari-pinned-tab.svg' color='#000000' />
-      <script src='https://i.wellcomecollection.org/assets/libs/picturefill.min.js' async />
-      {/* Leaving this out for now as it's hanging locally for me */}
-      {/* <script src='//platform.twitter.com/widgets.js' async defer></script> */}
-      <NastyJs />
-      <script type='application/ld+json'>{/* JSON+LD Z */}</script>
-      <script dangerouslySetInnerHTML={{ __html: `
-      window.WC = {
-        featuresCohort: ${JSON.stringify(featuresCohort)},
-        featureFlags: ${JSON.stringify(featureFlags)}
-      }
-    `}} />
-      {url && <link rel='canonical' href={url} />}
-    </Head>
+  componentDidUpdate() {
```

Comment:
Assuming we assume that this get's updated every time there's a change to the page?
If so - is there not a more explicit way to do this as I couldn't tell you what the condition is for this to update?

Reply:
Yeah, not sure. Really we want to run this every time the URL changes. Another option is to add something that listens for React route changes in the defaultPageLayout `componentDidMount()` method (or something similar).

Don't know what best practice is for this, though (or if doing it like this isn't sufficient).

Reply:
I would probably plug this into the next router?
https://github.com/zeit/next.js#router-events